unity/src/test/scala/io/delta/unity/UCCatalogManagedClientSuite.scala

```diff
@@ -78,37 +82,58 @@ class UCCatalogManagedClientSuite extends AnyFunSuite with UCCatalogManagedTestU
     val ucCatalogManagedClient = new UCCatalogManagedClient(ucClient)
 
     assertThrows[NullPointerException] {
-      ucCatalogManagedClient.loadTable(null, "ucTableId", "tablePath", 0L) // engine is null
+      // engine is null
+      ucCatalogManagedClient.loadTable(null, "ucTableId", "tablePath", Optional.of(0L))
     }
     assertThrows[NullPointerException] {
-      ucCatalogManagedClient.loadTable(defaultEngine, null, "tablePath", 0L) // ucTableId is null
+      // ucTableId is null
+      ucCatalogManagedClient.loadTable(defaultEngine, null, "tablePath", Optional.of(0L))
     }
     assertThrows[NullPointerException] {
-      ucCatalogManagedClient.loadTable(defaultEngine, "ucTableId", null, 0L) // tablePath is null
+      // tablePath is null
+      ucCatalogManagedClient.loadTable(defaultEngine, "ucTableId", null, Optional.of(0L))
     }
     assertThrows[IllegalArgumentException] {
-      ucCatalogManagedClient.loadTable(defaultEngine, "ucTableId", "tablePath", -1L) // version < 0
+      // version < 0
+      ucCatalogManagedClient.loadTable(defaultEngine, "ucTableId", "tablePath", Optional.of(-1L))
     }
   }
 
+  test("table version 0 is loaded when UC maxRatifiedVersion is -1") {
+    val tablePath = getTestResourceFilePath("catalog-owned-preview")
+    val ucClient = new InMemoryUCClient("ucMetastoreId")
+    val tableData = new TableData(-1, ArrayBuffer[Commit]())
+    ucClient.createTableIfNotExistsOrThrow("ucTableId", tableData)
+    val ucCatalogManagedClient = new UCCatalogManagedClient(ucClient)
+    val resolvedTable = ucCatalogManagedClient
+      .loadTable(defaultEngine, "ucTableId", tablePath, Optional.empty())
+      .asInstanceOf[ResolvedTableInternal]
+
+    assert(resolvedTable.getVersion == 0L)
+  }
+
   test("loadTable throws if version to load is greater than max ratified version") {
     val exMsg = intercept[IllegalArgumentException] {
-      testCatalogManagedTable(versionToLoad = 9L)
+      testCatalogManagedTable(versionToLoad = Optional.of(9L))
     }.getMessage
 
     assert(exMsg.contains("Cannot load table version 9 as the latest version ratified by UC is 2"))
   }
 
+  test("loadTable correctly laods a UC table -- versionToLoad is empty => load latest") {
+    testCatalogManagedTable(versionToLoad = Optional.empty())
```

Comment:
nit: its not obvious to me that this loads the latest, should it be asserted in the test?